docs/external-node.md

```diff
@@ -0,0 +1,565 @@
+
+# ExternalNode
+
+## Table of Contents
+
+<!-- toc -->
+- [What is ExternalNode?](#what-is-externalnode)
+- [Prerequisites](#prerequisites)
+- [The ExternalNode resource](#the-externalnode-resource)
+  - [Name and Namespace](#name-and-namespace)
+  - [Interfaces](#interfaces)
+- [Install Antrea Agent on VM](#install-antrea-agent-on-vm)
+  - [Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster)
+  - [Installation on Linux VM](#installation-on-linux-vm)
+    - [Prerequisites on Linux VM](#prerequisites-on-linux-vm)
+    - [Installation steps on Linux VM](#installation-steps-on-linux-vm)
+  - [Installation on Windows VM](#installation-on-windows-vm)
+    - [Prerequisites on Windows VM](#prerequisites-on-windows-vm)
+    - [Installation steps on Windows VM](#installation-steps-on-windows-vm)
+- [VM network configuration](#vm-network-configuration)
+- [RBAC for antrea-agent](#rbac-for-antrea-agent)
+- [Apply Antrea NetworkPolicy to ExternalNode](#apply-antrea-networkpolicy-to-externalnode)
+  - [Antrea NetworkPolicy configuration](#antrea-networkpolicy-configuration)
+  - [Bypass Antrea NetworkPolicy](#bypass-antrea-networkpolicy)
+- [OpenFlow pipeline](#openflow-pipeline)
+  - [Non-IP packet](#non-ip-packet)
+  - [IP packet](#ip-packet)
+- [Limitations](#limitations)
+<!-- /toc -->
+
+## What is ExternalNode?
+
+`ExternalNode` is a CRD API that enables Antrea to manage the network connectivity
+and security on a Non-Kubernetes Node (like a virtual machine or a bare-metal
+server). It supports specifying which network interfaces on the external Node
+are expected to be protected with Antrea NetworkPolicy rules. The virtual machine
+or bare-metal server represented by an `ExternalNode` resource can be either
+Linux or Windows. An "external Node" will be used to represent such a virtual
```

Comment:
Linux or Windows. "external Node" will be used to designate such a virtual machine or bare-metal server in the rest of this document.